dm/dm/master/scheduler/scheduler.go

```diff
@@ -1039,6 +1034,73 @@ func (s *Scheduler) RemoveSubTasks(task string, sources ...string) error {
 	return nil
 }
 
+// UpdateSubTasks update the information of one or more subtasks for one task.
+func (s *Scheduler) UpdateSubTasks(ctx context.Context, cfgs ...config.SubTaskConfig) error {
```

Comment:
what's the promise of concurrent call or atomicity?

Reply:
Concurrent requests are not allowed here
add lock in https://github.com/pingcap/tiflow/pull/4760/commits/73797989a5d006110455dec6f356e3ca856942f1